### PR TITLE
Deltalake UI

### PR DESCRIPTION
**Pull request by Karakatiza666**
Is this a user-visible change (yes/no): yes

Fix https://github.com/feldera/feldera/issues/1584 : Delta Lake output connector UI
Fix https://github.com/feldera/feldera/issues/1591 : Output buffer configuration UI
Fix https://github.com/feldera/feldera/issues/1744 : Delta Lake input adapter UI

As a part of this PR I refactored most connector dialogs to use the same layout as a component, rather than duplicating markup for the layout in each dialog (the new components that contain the layout are ConnectorEditDialog, PlainDialogContent, VerticalTabsDialogContent).

### CODE REVIEW

**Comment by ryzhyk** [comment]
The new connector doesn't appear in the input or output connector list
image

**Comment by ryzhyk** [comment]
For me tooltips only show up in the INGEST MODE tab. Not a big issue, just checking if they are meant to work.

**Comment by ryzhyk** [comment]
Can we make snapshot_version and snapshot datetime fields mutually exclusive (at most one is allowed to be specified)

**Comment by ryzhyk** [comment]
Can `aws_access_key_id` and other secret fields be covered with `****` unless the user chooses to reveal it?

**Comment by ryzhyk** [comment]
Some options are synonyms, and we need to keep one of them only:
- aws_access_key_id, access_key_id
- aws_secret_access_key, secret_access_key
- aws_region, region

See full list here: https://docs.rs/object_store/0.10.1/object_store/aws/enum.AmazonS3ConfigKey.html

**Comment by ryzhyk** [comment]
For output buffer config, when the buffer is enabled, can we force the user to set one of `max_output_buffer_time_millis` or `max_output_buffer_size_records`?

**Comment by ryzhyk** [comment]
`s3://feldera-fraud-detection-demo/demographics_inference` is recognized as S3-compatible URL, but `s3://feldera-fraud-detection-demo/` isn't

**Comment by ryzhyk** [comment]
`Update` in the input connector edit dialog does not have any effect:
update

**Comment by Karakatiza666** [comment]
> For me tooltips only show up in the INGEST MODE tab. Not a big issue, just checking if they are meant to work.
Only added them there

Addressed or fixed other comments

**Comment by ryzhyk** [comment]
- I still cannot click on Update
- `timestamp_column` option is missing in the input connector UI (see #1744)
- no `mode` option in the output connector UI - this one was added recently and I forgot to update the UI issue.  Here are the docs for it: https://github.com/feldera/feldera/blob/a3bc757045f2485df1b06ca20014686e7ec3d049/crates/pipeline-types/src/transport/delta_table.rs#L10
- output buffer should be enabled by default for the output delta connector (see #1591)
- output buffer dialog should require setting one of the limits, but it instead it prevents setting either of them

**Comment by ryzhyk** [comment]
After I save the connector and open it, `timestamp_column` and `snapshot_filter` appear in the Options tab:

image

**Comment by ryzhyk** [comment]
Let's not extract aws_bucket and aws_endpoint fields from the S3 URL. This is redundant, and causes the object store crate to panic (I'll report the issue to them)

**Comment by Karakatiza666** [comment]
@ryzhyk please help reproducing your issue of not being able to update the connector - e.g. connector configuration, and what do you change in form fields

**Comment by gz** [comment]
- This should say "Enter option name, press Enter to add it"
Screenshot from 2024-06-07 09-30-28

- some spaces missing in the text here:
Screenshot from 2024-06-07 09-33-03

- Clarify that snapshot version should be a number


- it doesn't seem good that 
```			"snapshot_filter": "",
			"timestamp_column": ""
```
remain in the JSON after we switch ingest mode to somethign that doesn't support it, or keep it as an empty option at all here

[Screencast from 06-07-2024 09:35:01 AM.webm](https://github.com/feldera/feldera/assets/127654/ef9f8da7-ad39-4bbd-b361-a260cce25508)


- it's not clear here what the options are: 
Screenshot from 2024-06-07 09-37-45

maybe best is to add a helper text as you do with the other settings under ingest mode

File: web-console/src/lib/services/manager/core/request.ts [hunk 1]
```diff
@@ -108,7 +110,7 @@ export const getFormData = (options: ApiRequestOptions): FormData | undefined =>
       if (isString(value) || isBlob(value)) {
         formData.append(key, value)
       } else {
-        formData.append(key, JSON.stringify(value))
+        formData.append(key, JSONbig.stringify(value))
```
Comment (gz) [inline web-console/src/lib/services/manager/core/request.ts:113]: why do we use JSONbig now?

File: web-console/src/lib/functions/deltalake/inferConfig.spec.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import { expect, test } from '@playwright/experimental-ct-react'
+
+import { inferDeltaLakeStorageConfig } from './inferConfig'
+
+// http://pkerschbaum.com/blog/using-playwright-to-run-unit-tests
+
+test('Test DeltaLake config inference', async ({}) => {
```
Comment (gz) [inline web-console/src/lib/functions/deltalake/inferConfig.spec.ts:7]: we should have more tests for this based on the issues reported by leonid e.g.,:

- json options should not just appear
- timestamp_column shouldn't remain in JSON when they are empty
- test that checks that you can't set both columns snapshotversion/snapshot delta

**Review by gz — APPROVED** [review]

**Comment by ryzhyk** [comment]
- Click on an input connector.
- It says "URI NOT RECOGNIZED"
- Click NEXT, and then PREVIOUS, it now says "AWS compatible URI"

uri_not_recognized

**Comment by ryzhyk** [comment]
When I leave the "timestamp column" field empty when creating a connector, it works correctly in that it doesn't send the field to the backend, but if I write something in that field, save the connector, and then edit the connector and clear the field, it gets saved as an empty string, which is wrong.

**Comment by ryzhyk** [comment]
From https://github.com/feldera/feldera/issues/1591 (re output buffer config):

> This feature should be disabled by default for all connectors, except delta table. For the latter, we need to choose a reasonable default, e.g., 10 seconds, 1M records.

When I create an output connector, output buffer is enabled, but not defaults are set:

image

**Review by ryzhyk — APPROVED** [review]

File: web-console/src/lib/services/manager/core/request.ts (hunk 1 above)
Comment (Karakatiza666) [reply web-console/src/lib/services/manager/core/request.ts:113]: The spec for max_output_buffer_time_millis and max_output_buffer_time_records fields dictates the type usize (u64), and default JSON serialization doesn't support 64-bit integers (it truncates them to a floating point number).
So here I am changing the auto-generated code for request serialization (a supported feature of openapi-typescript codegen) to parse wire JSON with a library that spits out BigNumber-s if it doesn't fit in a built-in JavaScript Number.

64-bit integers will probably get used elsewhere in API, so this change will be more useful as the time goes on.

**Pull request merged**